website/content/api-docs/secret/identity/mfa/duo.mdx

```diff
@@ -0,0 +1,159 @@
+| :----- |:-------------------------------|
+| `POST` | `/identity/mfa/method/duo/:id` |
+
+### Parameters
+
+- `id` `(string: "")` - Optional UUID to specify if updating an existing method.
+
+- `mount_accessor` `(string: <required>)` - The mount to tie this method to for use in automatic mappings. The mapping will use the Name field of Aliases associated with this mount as the username in the mapping.
+
+- `username_format` `(string)` - A format string for mapping Identity names to MFA methods. Values to substitute should be placed in `{{}}`. For example, `"{{alias.name}}@example.com"`. If blank, the Alias's Name field will be used as-is. Currently-supported mappings:
+
+  - alias.name: The name returned by the mount configured via the `mount_accessor` parameter
+  - entity.name: The name configured for the Entity
+  - alias.metadata.`<key>`: The value of the Alias's metadata parameter
+  - entity.metadata.`<key>`: The value of the Entity's metadata parameter
+
+- `secret_key` `(string)` - Secret key for Duo.
+
+- `integration_key` `(string)` - Integration key for Duo.
+
+- `api_hostname` `(string)` - API hostname for Duo.
+
+- `push_info` `(string)` - Push information for Duo.
+-
+- `use_passcode` `(bool: false)` - If true, the user is reminded to use the passcode upon MFA validation.
+
+### Sample Payload
+
+```json
+{
+  "mount_accessor": "auth_userpass_1793464a",
+  "secret_key": "BIACEUEAXI20BNWTEYXT",
+  "integration_key": "8C7THtrIigh2rPZQMbguugt8IUftWhMRCOBzbuyz",
+  "api_hostname": "api-2b5c39f5.duosecurity.com"
+}
+```
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/identity/mfa/method/duo
+```
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/identity/mfa/method/duo/4194659f-139b-400b-b5dd-86bfb726759d
+```
+
+## Read Duo MFA Method
+
+This endpoint queries the MFA configuration of Duo type for a given method
+ID.
+
+| Method | Path                           |
+| :----- |:-------------------------------|
+| `GET`  | `/identity/mfa/method/duo/:id` |
+
+### Parameters
+
+- `id` `(string: <required>)` – UUID of the MFA method.
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request GET \
+    http://127.0.0.1:8200/v1/identity/mfa/method/duo/4194659f-139b-400b-b5dd-86bfb726759d
+
+```
+
+### Sample Response
+
+```json
+{
+  "data": {
+    "api_hostname": "api-2b5c39f5.duosecurity.com",
+    "id": "4194659f-139b-400b-b5dd-86bfb726759d",
+    "integration_key": "BIACEUEAXI20BNWTEYXT",
+    "mount_accessor": "auth_userpass_1793464a",
+    "pushinfo": "",
+    "secret_key": "8C7THtrIigh2rPZQMbguugt8IUftWhMRCOBzbuyz",
+    "type": "duo",
+    "username_format": "",
+    "use_passcode": false
+  }
+}
+```
+
+## Delete Duo MFA Method
+
+This endpoint deletes a Duo MFA method.
```

Comment:
would it be worth it to mention that one can delete an MFA method only if the method is not used in any login enforcement configuration?

Reply:
Yeah, good idea.